experiments/kompanion/cmd/migrator/migrator.go

```diff
@@ -0,0 +1,442 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return RunMigrator(cmd.Context(), opts)
+		},
+		Args: cobra.ExactArgs(0),
+	}
+
+	flags := cmd.Flags()
+	opts.AddFlags(flags)
+
+	return cmd
+}
+
+// Task is implemented by our namespace-collection routine, or anything else we want to run in parallel.
+type Task interface {
+	Run(ctx context.Context) error
+}
+
+// tracks the namespaces to be exported.
+// thread safe.
+type taskQueue struct {
+	mu    sync.Mutex
+	tasks []Task // will be treated as a FIFO queue
+}
+
+func (n *taskQueue) GetWork() Task {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	if len(n.tasks) == 0 {
+		return nil
+	}
+
+	workItem := n.tasks[0]
+	n.tasks = n.tasks[1:]
+
+	return workItem
+}
+
+func (n *taskQueue) AddTask(t Task) {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	n.tasks = append(n.tasks, t)
+}
+
+type dumpResourcesTask struct {
+	// Namespace is the namespace to filter down
+	Namespace string
+
+	DynamicClient *dynamic.DynamicClient
+
+	OutputChannel chan string
+
+	// Resources is the list of resources to query
+	Resources []schema.GroupVersionResource
+
+	//ReportDir string
+
+	ShouldExcludeObject func(id types.NamespacedName) bool
+}
+
+func (t *dumpResourcesTask) Run(ctx context.Context) error {
+	// log := klog.FromContext(ctx)
+
+	for _, gvr := range t.Resources {
+		var resources *unstructured.UnstructuredList
+		if t.Namespace != "" {
+			r, err := t.DynamicClient.Resource(gvr).Namespace(t.Namespace).List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return fmt.Errorf("fetching gvr %s resources in namespace %s: %w", gvr, t.Namespace, err)
+			}
+			resources = r
+
+		} else {
+			r, err := t.DynamicClient.Resource(gvr).List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return fmt.Errorf("fetching gvr %s resources: %w", gvr, err)
+			}
+			resources = r
+		}
+
+		for _, r := range resources.Items {
+			id := types.NamespacedName{
+				Namespace: r.GetNamespace(),
+				Name:      r.GetName(),
+			}
+			if t.ShouldExcludeObject(id) {
+				continue
+			}
+			r = pruneResource(r)
+
+			data, err := yaml.Marshal(r)
+			if err != nil {
+				return fmt.Errorf("error marshalling resource %s: %w", id, err)
+			}
+			t.OutputChannel <- string(data)
+			/*
+				filename := filepath.Join(t.ReportDir, id.Namespace, fmt.Sprintf("%s_%s.yaml", r.GroupVersionKind().Kind, id.Name))
+				filedir := filepath.Dir(filename)
+				if err := os.MkdirAll(filedir, 0o700); err != nil {
+					return fmt.Errorf("error creating directory %s: %w", filedir, err)
+				}
+
+				if err := os.WriteFile(filename, data, 0o644); err != nil {
+					return fmt.Errorf("error writing file for resource %s in namespace %s: %w", id.Name, id.Namespace, err)
+				}
+			*/
+		}
+	}
+
+	return nil
+}
+
+func pruneNamespace(ns v1.Namespace) v1.Namespace {
+	newAnnotations := make(map[string]string)
+	annotations := ns.GetAnnotations()
+	for key, value := range annotations {
+		if !strings.Contains(key, "cnrm.cloud.google.com") {
```

Comment:
I think we want to keep the project-id annotation here (?).  But what are the other annotations that we are removing?